tajo-core/src/main/java/org/apache/tajo/engine/function/datetime/DatePartFromTimestamp.java

```diff
@@ -59,8 +55,9 @@ public DatePartFromTimestamp() {
 
   @Override
   public void init(OverridableConf context, FunctionEval.ParamType [] types) {
-    String timezoneId = context.get(SessionVars.TIMEZONE, TajoConstants.DEFAULT_SYSTEM_TIMEZONE);
-    timezone = TimeZone.getTimeZone(timezoneId);
+    if (!hasTimeZone()) {
```

Comment:
It looks a bug to me. As you know, the reason why the results of the above queries are different is there is no way to apply the client timezone to the result of date_part(). This is also same for add_month() and add_day() functions. I think we need to fix this too because the result is not the one which users usually expect. Of course, the result is different from that of pgsql.

But, this also looks to require a huge amount of work. If you agree, how about handling this issue in another Jira?